std/numeric.d

```diff
@@ -1400,6 +1400,228 @@ unittest
     static assert(__traits(compiles, findRoot((real x)=>cast(double)x, real.init, real.init)));
 }
 
+/++
+Find a real minimum of a real function $(D f(x)) via bracketing.
+Given a function $(D f) and a range $(D (ax..bx)),
+returns the value of $(D x) in the range which is closest to a minimum of $(D f(x)).
+$(D f) is never evaluted at the endpoints of $(D ax) and $(D bx).
+If $(D f(x)) has more than one minimum in the range, one will be chosen arbitrarily.
+If $(D f(x)) returns NaN or -Infinity, $(D (x, f(x), NaN)) will be returned;
+otherwise, this algorithm is guaranteed to succeed.
+
+Params:
+    f = Function to be analyzed
+    ax = Left bound of initial range of f known to contain the minimum.
+    bx = Right bound of initial range of f known to contain the minimum.
+    relTolerance = Relative tolerance.
+    absTolerance = Absolute tolerance.
+
+Preconditions:
+    $(D ax) and $(D bx) shall be finite reals. $(BR)
+    $(D relTolerance) shall be normal positive real. $(BR)
+    $(D absTolerance) shall be normal positive real no less then $(D T.epsilon*2).
+
+Returns:
+    A tuple consisting of $(D x), $(D y = f(x)) and $(D error = 3 * (absTolerance * fabs(x) + relTolerance)).
+
+    The method used is a combination of golden section search and
+successive parabolic interpolation. Convergence is never much slower
+than that for a Fibonacci search.
+
+References:
+    "Algorithms for Minimization without Derivatives", Richard Brent, Prentice-Hall, Inc. (1973)
+
+See_Also: $(LREF findRoot), $(XREF math, isNormal)
++/
+Tuple!(T, "x", Unqual!(ReturnType!DF), "y", T, "error")
+findLocalMin(T, DF)(
+        scope DF f,
+        in T ax,
+        in T bx,
+        in T relTolerance = sqrt(T.epsilon),
+        in T absTolerance = sqrt(T.epsilon),
+        )
+    if (isFloatingPoint!T
+        && __traits(compiles, {T _ = DF.init(T.init);}))
+in
+{
+    assert(isFinite(ax), "ax is not finite");
+    assert(isFinite(bx), "bx is not finite");
+    assert(isNormal(relTolerance), "relTolerance is not normal floating point number");
+    assert(isNormal(absTolerance), "absTolerance is not normal floating point number");
+    assert(relTolerance >= 0, "absTolerance is not positive");
+    assert(absTolerance >= T.epsilon*2, "absTolerance is not greater then `2*T.epsilon`");
+}
+out (result)
+{
+    assert(isFinite(result.x));
+}
+body
+{
+    alias R = Unqual!(CommonType!(ReturnType!DF, T));
+    // c is the squared inverse of the golden ratio
+    // (3 - sqrt(5))/2
+    // Value obtained from Wolfram Alpha.
+    immutable T c = 0x0.61c8864680b583ea0c633f9fa31237p+0L;
+    immutable T cm1 = 0x0.9e3779b97f4a7c15f39cc0605cedc8p+0L;
```

Comment:
Would you lose anything by replacing these with the following?

    enum c = cast(T) (3.0L - sqrt(5.0L))/2.0L;
    enum cm1 = c - 1;

Then the values are self explanatory, no need for comments and magic numbers.